tests/test_taskmanager.py

```diff
@@ -0,0 +1,156 @@
+# -*- coding: utf-8 -*-
+
+import django
+import unittest2 as unittest
+
+# Only for Django >= 1.7
+if 'setup' in dir(django):
+    # This has to happen before below imports because they have a hard requirement
+    # on settings being loaded before import.
+    django.setup()
+
+from django.contrib.auth.models import User
+
+from mock import patch
+
+from kobo.client.constants import TASK_STATES
+from kobo.exceptions import ShutdownException
+from kobo.hub.models import Arch, Channel, Task, Worker
+from kobo.worker import TaskBase
+from kobo.worker.taskmanager import TaskManager, TaskContainer
+
+from .rpc import HubProxyMock, RpcServiceMock
+from .utils import DjangoRunner
+
+runner = DjangoRunner()
+setup_module = runner.start
+teardown_module = runner.stop
+
+
+class DummyTask(TaskBase):
+    enabled = True
+    arches = ['testarch']
+    channels = ['testchannel']
+    exclusive = False
+    foreground = True
+    priority = 10
+    weight = 1.0
+
+    def run(self):
+        self.result = True
+
+    @classmethod
+    def cleanup(cls, hub, conf, task_info):
+        pass
+
+    @classmethod
+    def notification(cls, hub, conf, task_info):
+        pass
+
+
+class TestTaskManager(unittest.TestCase):
+
+    def setUp(self):
+        TaskContainer.register_plugin(DummyTask)
+
+        user = User.objects.create()
+        arch = Arch.objects.create(name='testarch')
+        channel = Channel.objects.create(name='testchannel')
+
+        w = Worker.objects.create(
+            worker_key='mock-worker',
+            name='mock-worker',
+        )
+
+        w.arches.add(arch)
+        w.channels.add(channel)
+        w.save()
+
+        self._arch = arch
+        self._channel = channel
+        self._user = user
+        self._worker = RpcServiceMock(w)
+
+    def tearDown(self):
+        Task.objects.all().delete()
```

Comment:
Is this needed?  I'd expect it shouldn't be, because part of the DjangoRunner's responsibility is to create a fresh DB for each test and delete it afterward.

Reply:
Django creates a fresh DB for each run, not for each test case. I don't want data from previous tests affecting others.

Reply:
so, if I use django.test.TestCase as a base class I think these deletes are not required.